### PR TITLE
[1gsD8V8g] Add a more useful error message for export functions

### PR DESCRIPTION
**Pull request by gem-neo4j**
When trying to use SHOW CONSTRAINTS or SHOW INDEXES, it looks like the current user doesn't have permission to use it instead of the procedure, this adds a more generic error message.

### CODE REVIEW

File: common/src/main/java/org/neo4j/cypher/export/CypherResultSubGraph.java [hunk 1]
```diff
@@ -44,12 +47,16 @@ public void add(Relationship rel) {
...
-        });
+        try {
+            result.forEachRemaining(row -> {
+                for (String column : columns) {
+                    final Object value = row.get(column);
+                    graph.addToGraph(value);
+                }
+            });
+        } catch (AuthorizationViolationException e) {
+            throw new RuntimeException(INVALID_QUERY_MODE_ERROR);
```
Comment (Lojjs) [inline common/src/main/java/org/neo4j/cypher/export/CypherResultSubGraph.java:58]: Besides `apoc.export.cypher.query` and some procedures which does not take queries,  this method is also used by `apoc.export.graphml.query`. Do we want to add a test for that one as well?

File: common/src/main/java/apoc/util/Util.java [hunk 2]
```diff
@@ -109,6 +109,9 @@ public class Util {
 
     public static final int REDIRECT_LIMIT = 10;
 
+    public static String INVALID_QUERY_MODE_ERROR = "This procedure allows for READ-only, non-schema based queries. " +
+            "It is therefore not possible to perform writes or query the database with commands such as SHOW CONSTRAINTS/INDEXES.";
```
Comment (Lojjs) [inline common/src/main/java/apoc/util/Util.java:113]: Might be worth adding a non-schema write query to each test as well, to make sure that the write part of the error message holds.

**Review by Lojjs — CHANGES_REQUESTED** [review]
I think the solution seems fine but there are some additions to testing I would like to see

File: core/src/test/java/apoc/load/LoadCoreSecurityTest.java [hunk 3]
```diff
@@ -144,11 +144,4 @@ public void testReadSensitiveFileWorks() {
             }
         }
     }
-
-    private static void assertError(Exception e, String errorMessage, Class<? extends Exception> exceptionType, String apocProcedure) {
```
Comment (Lojjs) [inline core/src/test/java/apoc/load/LoadCoreSecurityTest.java:148]: I think this removal has caused some unused imports

File: common/src/main/java/org/neo4j/cypher/export/CypherResultSubGraph.java (hunk 1 above)
Comment (gem-neo4j) [reply common/src/main/java/org/neo4j/cypher/export/CypherResultSubGraph.java:58]: Ah I missed one! nice catch :)

File: common/src/main/java/apoc/util/Util.java (hunk 2 above)
Comment (gem-neo4j) [reply common/src/main/java/apoc/util/Util.java:113]: The writes are actually caught in a different place, but I liked the more generic message which is why I wrote it like this, as it still true you can't write in any of these queries

**Review by Lojjs — APPROVED** [review]

**Pull request merged**